### PR TITLE
Feature/orca 175

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-175: Create and/or modify helper lambda to translate Cumulus Dashboard inputs into the proper schema input for copy_to_glacier](https://bugs.earthdata.nasa.gov/browse/ORCA-175)

## Changes

* Added lambda for translating CumulusDashboard input to copy_to_glacier format

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests (unit, integration, ad-hoc, or otherwise) that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] New unit tests.
- [x] Tested manually in AWS.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
    - [x] Development documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
    - [x] Manual testing/integration testing passes (if forked)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

## What is the current behavior?

(What actually happens)

## What is the expected correct/added behavior?

(What you should see instead)

## Relevant logs and/or screenshots

(Paste any relevant logs - please use code blocks (```) to format console output,
logs, and code as it's very hard to read otherwise.)

## Possible fixes

(If you can, link to the line of code that might be responsible for the problem)

### CODE REVIEW

File: tasks/copy_to_glacier_cumulus_translator/copy_to_glacier_cumulus_translator.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    Args:
+        event: A dict with the following keys:
+            'input': See schemas/input.json
+            'config': See schemas/config.json
+        context: An object passed through by CMA. Unused.
+
+    Returns:
+        See schemas/output.json
+    """
+    # Cannot use f"" because of '{}' handling bug in CumulusLogger
```
Comment (rizbi-hassan) [inline tasks/copy_to_glacier_cumulus_translator/copy_to_glacier_cumulus_translator.py:27]: have we created  a card for fixing this bug for Cumulus team? @bhazuka

File: modules/lambdas/main.tf [hunk 2]
```diff
@@ -72,8 +72,36 @@ module "restore_object_arn" {
...
-# copy_to_glacier - Copies files to the ORCA S3 Glacier bucket
+# copy_to_glacier_cumulus_translator - Transforms input from Cumulus Dashboard to copy_to_glacier format
 # ==============================================================================
+resource "aws_lambda_function" "copy_to_glacier_cumulus_translator" {
+  ## REQUIRED
+  function_name = "${var.prefix}_copy_to_glacier_cumulus_translator"
+  role          = module.restore_object_arn.restore_object_role_arn
+
+  ## OPTIONAL
+  description      = "ORCA archiving lambda used to copy data to an ORCA S3 glacier bucket."
```
Comment (bhazuka) [inline modules/lambdas/main.tf:83]: Update the description for the Lambda please.

File: tasks/copy_to_glacier_cumulus_translator/copy_to_glacier_cumulus_translator.py (hunk 1 above)
Comment (bhazuka) [reply tasks/copy_to_glacier_cumulus_translator/copy_to_glacier_cumulus_translator.py:27]: I have not yet. Talking with the team on slack, we can submit a pull request and update the code ourselves for a release.

File: tasks/copy_to_glacier_cumulus_translator/schemas/input.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"$id": "https://github.com/nasa/cumulus-orca/blob/master/tasks/copy_to_glacier_cumulus_translator/schemas/input.json",
+	"title": "copy_to_glacier_cumulus_translator Lambda Input",
+	"description": "The input for the copy_to_glacier_cumulus_translator Lambda.",
+	"type": "object",
+	"required": [
+		"granules"
+	],
+	"properties": {
```
Comment (bhazuka) [inline tasks/copy_to_glacier_cumulus_translator/schemas/input.json:10]: You will also need to make sure some collection configuration information gets passed in as well from the CMA. Namely, the information related to the excluded file types and the regex configuration for the files I believe that copy_to_glacier uses to determine what files to archive and the key.

**Review by rizbi-hassan — APPROVED** [review]

File: tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+        mock_run_cumulus_task.assert_called_once_with(
+            copy_to_glacier_cumulus_translator.task, event, context
+        )
+        self.assertEqual(mock_run_cumulus_task.return_value, result)
+
+    @patch("copy_to_glacier_cumulus_translator.task")
+    @patch("cumulus_logger.CumulusLogger.setMetadata")
+    def test_handler_passthrough(
+        self, mock_setMetadata: MagicMock, mock_task: MagicMock
+    ):
```
Comment (bhazuka) [inline tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py:45]: Quick doc string on test would be good here.

File: tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+                        "granuleId": "some_granule_id",
+                    }
+                ]
+            },
+            result["payload"],
+        )
+
+    # noinspection PyPep8Naming
+    @patch("cumulus_logger.CumulusLogger.setMetadata")
+    def test_handler_rejects_bad_input(self, mock_setMetadata: MagicMock):
```
Comment (bhazuka) [inline tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py:123]: Quick doc string on test would be good here.

File: tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+                }
+            ]
+        }
+        event = {"task_config": expected_config, "payload": expected_input}
+        context = Mock()
+
+        with self.assertRaises(jsonschema.exceptions.ValidationError):
+            copy_to_glacier_cumulus_translator.handler(event, context)
+
+    def test_task_happy_path(self):
```
Comment (bhazuka) [inline tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py:153]: Quick doc string on test would be good here.

**Review by bhazuka — CHANGES_REQUESTED** [review]
Overall looks good, couple of questions and a few minor additions and changes to consider.

File: tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py (hunk 4 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py:45]: Added

File: modules/lambdas/main.tf (hunk 2 above)
Comment (AndrewDorn) [reply modules/lambdas/main.tf:83]: Whoops. Fixed.

File: tasks/copy_to_glacier_cumulus_translator/copy_to_glacier_cumulus_translator.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    LOGGER.debug("event: {event}", event=event)
+    file_name_mapping = event["config"]["file_mapping"]["name"]
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/copy_to_glacier_cumulus_translator/copy_to_glacier_cumulus_translator.py:29]: Not according to the example we were given. I think you're referring to the "task_config" in my tests, yes? That is auto-magically transformed by CMA.

File: CHANGELOG.md [hunk 8]
```diff
@@ -15,6 +15,8 @@ and includes an additional section for migration notes.
 
 
 ## [Unreleased]
+### Added
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply CHANGELOG.md:18]: Fixed

File: tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py (hunk 6 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py:156]: Added

File: tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py (hunk 5 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier_cumulus_translator/test/unit_tests/test_copy_to_glacier_cumulus_translator_unit.py:126]: Added

File: tasks/copy_to_glacier_cumulus_translator/schemas/input.json (hunk 3 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier_cumulus_translator/schemas/input.json:10]: Looks like that comes from the config, which is defined in a workflow on a per-lambda basis. Am I misunderstanding something?

Comment (bhazuka) [reply tasks/copy_to_glacier_cumulus_translator/schemas/input.json:10]: ok, I wasn't sure if that was needed as part of your JSON description or not.

**Review by bhazuka — APPROVED** [review]

**Pull request merged**